### PR TITLE
CNCF Self Assessment: Rook

### PR DESCRIPTION
**Pull request by dhauss**
First draft of [Rook](https://github.com/rook/rook) Project Security Self-Assessment

### CODE REVIEW

File: assessments/projects/rook/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+security of the project, determining gaps in their security, and preparing any security
+documentation for their users. This document is ideal for projects currently in the
+CNCF **sandbox** as well as projects that are looking to receive a joint assessment and
+currently in CNCF **incubation**.
+
+For a detailed guide with step-by-step discussion and examples, check out the free 
+Express Learning course provided by Linux Foundation Training & Certification: 
+[Security Assessments for Open Source Projects](https://training.linuxfoundation.org/express-learning/security-self-assessments-for-open-source-projects-lfel1005/).
+
+# Self-assessment outline
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:12]: All web content should only have a single H1 entry (#)

File: assessments/projects/rook/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
+# Self-assessment
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:1]: Please note the project name in the H1 heading, like # Rook Self-asssesment

File: assessments/projects/rook/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+| -- | -- |
+| Software |  [Rook](https://github.com/rook/rook)  |
+| Security Provider | No  |
+| Languages | Go, Python, C++ |
+| SBOM | [prerequisites](https://github.com/rook/rook/blob/master/Documentation/Getting-Started/Prerequisites/prerequisites.md), [go.mod](https://github.com/rook/rook/blob/master/pkg/apis/go.mod), [API go.mod](https://github.com/rook/rook/blob/master/pkg/apis/go.mod), [Build Reqs](https://github.com/rook/rook/blob/master/INSTALL.md), [images](https://github.com/rook/rook/blob/v1.12.9/deploy/examples/images.txt)|
+| | |
+
+### Security links
+
+Provide the list of links to existing security documentation for the project. You may
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:45]: Please remove any generic content from the template like line 45-46.

File: assessments/projects/rook/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+A table at the top for quick reference information, later used for indexing.
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:33]: Please remove any generic content from the template

File: assessments/projects/rook/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,374 @@
...
+Rook is an open source **cloud-native storage orchestrator** for Kubernetes, providing the platform, framework, and support for Ceph storage to natively integrate with Kubernetes.
+
+[Ceph](https://ceph.com/) is a distributed storage system that provides file, block and object storage and is deployed in large scale production clusters.
+
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+The status of the Ceph storage provider is **Stable**. Features and improvements will be planned for many future versions. Upgrades between versions are provided to ensure backward compatibility between releases.
+
+### Actors
+These are the individual parts of your system that interact to provide the
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:67]: Please remove any generic content from the template.

File: assessments/projects/rook/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+Rook automatically configures the Ceph-CSI driver to mount the storage to your pods. The rook/ceph image includes all necessary tools to manage the cluster. Rook is not in the Ceph data path. Many of the Ceph concepts like placement groups and crush maps are hidden so you don't have to worry about them. Instead, Rook creates a simplified user experience for admins that is in terms of physical resources, pools, volumes, filesystems, and buckets. Advanced configuration can be applied when needed with the Ceph tools.
+
+#### `Ceph CSI Driver`
+The Ceph-CSI driver provides the provisioning and mounting of volumes
+
+#### `Ceph Daemons`
+The Ceph daemons run the core storage architecture. See the [Glossary](https://github.com/rook/rook/blob/master/Documentation/Getting-Started/glossary.md#ceph) to learn more about each daemon.
+
+### Actions
+These are the steps that a project performs in order to provide some service
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:96]: This section is meant to hold the possible actions in this project. 
Please remove any generic content from the template, and refer to this text as a guide to add relevant content.

File: assessments/projects/rook/self-assessment.md [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+data, and interactions between actors to perform an action.  
+
+For example, the access server receives the client request, checks the format, 
+validates that the request corresponds to a file the client is authorized to 
+access, and then returns a token to the client.  The client then transmits that 
+token to the file server, which, after confirming its validity, returns the file.
+
+Rook can be used to automatically configure the Ceph CSI drivers to mount the storage to an application's pods. See image at top of [Actors](#actors) for reference
+
+##### Actors
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:109]: Kindly elaborate on the roles of each of these actors. A good example is [Actors - Pixie Self assessment](https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#actors)

File: assessments/projects/rook/self-assessment.md [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+Rook automates deployment and management of Ceph to provide self-managing, self-scaling, and self-healing storage services. The Rook operator does this by building on Kubernetes resources to deploy, configure, provision, scale, upgrade, and monitor Ceph.
+
+#### Security Goals
+
+* All access to rook operator should be authenticated and authorized
+* Secrets created by Rook operator should maintain confidentiality
+* Rook operator should obey the principle of least privilege (see [here](https://github.com/rook/rook/pull/1736) for improvements already made to operator permissions model)
+* Ceph storage elements should maintain integrity and availability while scaling
+
+### Non-goals
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:166]: This section is meant to outline what aren't the goals of the project. It isn't required all of them are security non-goals.
A good example is [Pixie non-goals](https://github.com/cncf/tag-security/blob/main/assessments/projects/pixie/self-assessment.md#non-goals)

File: assessments/projects/rook/self-assessment.md [hunk 9]
```diff
@@ -0,0 +1,374 @@
...
+#### Automated Checks
+
+Rook includes automated checks for vulnerabilities. These checks are part of Rook's continuous integration (CI) process and automatically run against every pull request. The results of these tests, along with code reviews and other criteria, determine whether a pull request will be accepted into the Rook repository
+
+#### Integration Tests
+
+Rook's upstream continuous integration (CI) tests will run integration tests against your changes automatically. You do not need to run these tests locally, but you may if you like.
+
+
+* Communication Channels.
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:244]: This statement might need to be formatted to be a heading

File: assessments/projects/rook/self-assessment.md [hunk 10]
```diff
@@ -0,0 +1,374 @@
...
+### Patching/Update Availability
+
+Once the vulnerability has been confirmed and the relevant parties have been notified, the next step is to make a patch or update available. This involves releasing a new version of the software that addresses the vulnerability. The patch or update is then made available to all users, who can then update their systems to the latest version to protect against the vulnerability.
+In addition, Rook has a Fix Disclosure Process that includes the disclosure of forthcoming fixes to users, which is completed within 1-7 days of disclosure. The Fix Lead will create a GitHub issue in the Rook project to inform users that a security vulnerability has been disclosed and that a fix will be made available, with an estimation of the release date
+
+## Incident Response
+
+There is a template for incident response for reference [here](https://github.com/cncf/tag-security/blob/main/project-resources/templates/incident-response.md)
+
+## Appendix
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:325]: Per your study and analysis, what would you recommend for enhancing the security of the project? Please outline here

**Review by ragashreeshekar — CHANGES_REQUESTED** [review]
Thanks for the PR @dhauss and team, appreciate the efforts.
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.
Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

File: assessments/projects/rook/self-assessment.md [hunk 11]
```diff
@@ -0,0 +1,374 @@
+# Self-assessment
+The Self-assessment is the initial document for projects to begin thinking about the
```
[the remark this answers is not in the document]
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Thank you for the prompt feedback! I removed the template text in this commit: [cfbb19f](https://github.com/dhauss/tag-security/commit/cfbb19f0ab420fbffd3366d72b2d3c037085bf8d)

File: assessments/projects/rook/self-assessment.md (hunk 2 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Fixed in this commit: [cfbb19f](https://github.com/dhauss/tag-security/commit/cfbb19f0ab420fbffd3366d72b2d3c037085bf8d)

File: assessments/projects/rook/self-assessment.md (hunk 3 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Most generic content removed in this commit: [cfbb19f](https://github.com/dhauss/tag-security/commit/cfbb19f0ab420fbffd3366d72b2d3c037085bf8d)

File: assessments/projects/rook/self-assessment.md (hunk 4 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md:24]: Same commit as above: [cfbb19f](https://github.com/dhauss/tag-security/commit/cfbb19f0ab420fbffd3366d72b2d3c037085bf8d)

File: assessments/projects/rook/self-assessment.md (hunk 6 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Generic content removed and actor section significantly expanded in this commit: [fe0d055](https://github.com/dhauss/tag-security/commit/fe0d055e96a6fa26851fc13a7d9d5ed33aa98ed4)

File: assessments/projects/rook/self-assessment.md (hunk 7 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md:99]: Generic content removed and actor section significantly expanded in this commit: [fe0d055](https://github.com/dhauss/tag-security/commit/fe0d055e96a6fa26851fc13a7d9d5ed33aa98ed4)

File: assessments/projects/rook/self-assessment.md (hunk 9 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Fixed in this quick commit (I'll be happy to squash these later, just want to keep the commits while we're still editing the project): [3774c00](https://github.com/dhauss/tag-security/commit/3774c006f129416e211a396c2858f0f774307b1e)

File: assessments/projects/rook/self-assessment.md (hunk 8 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md:156]: Expanded here: [ee390db](https://github.com/dhauss/tag-security/commit/ee390db799dfd7245bf65fa995675fe746123939)

File: assessments/projects/rook/self-assessment.md (hunk 5 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Generic content removed in this commit: [fe0d055](https://github.com/dhauss/tag-security/commit/fe0d055e96a6fa26851fc13a7d9d5ed33aa98ed4)

File: assessments/projects/rook/self-assessment.md (hunk 1 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: removed second h1 entry here: [ee390db](https://github.com/dhauss/tag-security/commit/ee390db799dfd7245bf65fa995675fe746123939)

File: assessments/projects/rook/self-assessment.md (hunk 6 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Expanded here: [5eb312b](https://github.com/dhauss/tag-security/commit/5eb312bff2c7ec72431122574e782f57036c42fd)

File: website/themes/docsy/userguide/package.json [hunk 12]
```diff
@@ -18,7 +18,7 @@
   "homepage": "https://github.com/bep/tech-doc-hugo#readme",
   "dependencies": {},
   "devDependencies": {
-    "autoprefixer": "^9.5.0",
```
Comment (ragashreeshekar) [inline website/themes/docsy/userguide/package.json:21]: Why is the change to website and themes folder being made? 
This doesn't seem to be relevant to the self-assessment and I'd suggest discarding this change

Comment (dhauss) [reply website/themes/docsy/userguide/package.json:21]: I think this happened when I synced our repo to the cncf main branch. Sorry for the confusion, I can checkout this file to an older commit. Looks similar to this dependabot [commit](https://github.com/cncf/tag-security/pull/1172/commits/96b4f6bbc7bd9fb052df5241a7b959d27f645fc5), but the autoprefixer commit was likely squashed

File: assessments/projects/rook/self-assessment.md [hunk 13]
```diff
@@ -0,0 +1,472 @@
+# Rook Self-assessment
+
+## Self-assessment outline
+
```
Comment (eddie-knight) [inline assessments/projects/rook/self-assessment.md:4]:
```suggestion
```

File: assessments/projects/rook/self-assessment.md (hunk 4 above)
Comment (eddie-knight) [reply assessments/projects/rook/self-assessment.md:24]: Still showing up on latest for me. Likely got restored by a later commit.

File: assessments/projects/rook/self-assessment.md [hunk 14]
```diff
@@ -0,0 +1,472 @@
...
+* [Self-assessment use](#self-assessment-use)
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+A table at the top for quick reference information, later used for indexing.
```
Comment (eddie-knight) [inline assessments/projects/rook/self-assessment.md:24]:
```suggestion
```

File: assessments/projects/rook/self-assessment.md [hunk 15]
```diff
@@ -0,0 +1,472 @@
...
+
+|   |  |
+| -- | -- |
+| Software |  [Rook](https://github.com/rook/rook)  |
+| Security Provider | No  |
+| Languages | Go, Python, C++ |
+| SBOM | [prerequisites](https://github.com/rook/rook/blob/master/Documentation/Getting-Started/Prerequisites/prerequisites.md), [go.mod](https://github.com/rook/rook/blob/master/pkg/apis/go.mod), [API go.mod](https://github.com/rook/rook/blob/master/pkg/apis/go.mod), [Build Reqs](https://github.com/rook/rook/blob/master/INSTALL.md), [images](https://github.com/rook/rook/blob/v1.12.9/deploy/examples/images.txt)|
```
Comment (eddie-knight) [inline assessments/projects/rook/self-assessment.md:31]: Recommend simplifying this to simply demonstrate that the target state is not met.

```suggestion
| SBOM | Rook does not currently generate an SBOM on release |
```

File: assessments/projects/rook/self-assessment.md (hunk 10 above)
Comment (eddie-knight) [reply assessments/projects/rook/self-assessment.md:423]: Ammending on Raga's last message, the review team is no longer asking for recommendations in this document (instead, we're asking that they _not_ be included).

File: assessments/projects/rook/self-assessment.md (hunk 4 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Honestly think I just missed it- removed it for real now: [this commit](https://github.com/dhauss/tag-security/commit/ab38ebc933b142aa6f432e7b7516f59a58975920)

File: assessments/projects/rook/self-assessment.md (hunk 14 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: fixed this one manually above ([commit](https://github.com/dhauss/tag-security/commit/ab38ebc933b142aa6f432e7b7516f59a58975920)), thanks for catching this

File: website/themes/docsy/userguide/package.json (hunk 12 above)
Comment (dhauss) [reply website/themes/docsy/userguide/package.json]: I reverted the website folder back in commit [7c1c8ae](https://github.com/dhauss/tag-security/commit/abb8f8e076bd44da45685ebffffaf429b6d4ab1a), I believe this should fix the issue but let me know if it didn't, admittedly my first time doing this particular git move

File: assessments/projects/rook/self-assessment.md (hunk 10 above)
Comment (dhauss) [reply assessments/projects/rook/self-assessment.md:419]: Thank you for the clarification, and thanks again to both of you for all the feedback

File: assessments/projects/rook/self-assessment.md [hunk 16]
```diff
@@ -0,0 +1,468 @@
...
+* [object-bucket-claim-retain.yaml](https://github.com/rook/rook/blob/master/deploy/examples/object-bucket-claim-retain.yaml) Creates a request for a new bucket by referencing a StorageClass which saves the bucket when the initiating OBC is deleted.
+* [object-bucket-claim-delete.yaml](https://github.com/rook/rook/blob/master/deploy/examples/object-bucket-claim-delete.yaml) Creates a request for a new bucket by referencing a StorageClass which deletes the bucket when the initiating OBC is deleted.
+* [storageclass-bucket-retain.yaml](https://github.com/rook/rook/blob/master/deploy/examples/storageclass-bucket-retain.yaml) Creates a new StorageClass which defines the Ceph Object Store and retains the bucket after the initiating OBC is deleted.
+* [storageclass-bucket-delete.yaml](https://github.com/rook/rook/blob/master/deploy/examples/storageclass-bucket-delete.yaml) Creates a new StorageClass which defines the Ceph Object Store and deletes the bucket after the initiating OBC is deleted.
+
+##### Actors
+
+* Rook Operator
+* Ceph CSI Drivers
+* Ceph Daemons
```
Comment (eddie-knight) [inline assessments/projects/rook/self-assessment.md:200]: What is the role of this H5 subsection, compared to the H3 Actors above? Consider removing this if it is not filling a purpose.

File: assessments/projects/rook/self-assessment.md [hunk 17]
```diff
@@ -0,0 +1,468 @@
...
+
+**Security**
+
+* Ceph- and K8s-specific vulnerabilities
+* Vulnerabilities caused by user error
+* Address security issues of extensions or tools used with Rook
+
+## Self-assessment use
+
+This self-assessment is created by the Rook team to perform an internal analysis of the
```
Comment (eddie-knight) [inline assessments/projects/rook/self-assessment.md:268]: Was this created with active participation and review from the Rook team? Consider listing the authors of this self-assessment in the first lines, below the title header.

Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Added authors and Rook level of participation [here](https://github.com/dhauss/tag-security/commit/f2318b659eac315a4b1e04001f2702eb467384e1). We had feedback from a maintainer, Travis Nielsen, but the Rook team was busy with a version update so his ability to participate was a bit limited

Comment (eddie-knight) [reply assessments/projects/rook/self-assessment.md]: Thanks @dhauss— Please adjust the phrasing to not state that this was created by the Rook team. The sentence may simply be omitted as you have the authorship documented at the top.

Comment (dhauss) [reply assessments/projects/rook/self-assessment.md]: Sounds good- I omitted the sentence, and also removed a reference to seeking graduation given Rook already graduated. You can find the changes in this [commit](https://github.com/dhauss/tag-security/commit/b89d1c4ac7670abb197293d7b8ef086d8ac63aef), happy to adjust further if needed

**Comment by dhauss** [comment]
Hey @ragashreeshekar and @eddie-knight , just double checking to see if there's anything pending I missed here? Also should I update the branch, or will that just bring back the unwanted changes to the website folder? Thank you!

**Comment by eddie-knight** [comment]
If there are no conflicts identified by github and the directory structure is unchanged, the branch rarely needs to be brought fully up to date.

File: assessments/projects/rook/self-assessment.md [hunk 18]
```diff
@@ -0,0 +1,465 @@
+# Rook Self-assessment
+
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:2]:
```suggestion
This Self-assessment document was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process, and is currently pending changes from the maintainer team. 

```

File: assessments/projects/rook/self-assessment.md [hunk 19]
```diff
@@ -0,0 +1,466 @@
...
+* [Security functions and features](#security-functions-and-features)
+* [Project compliance](#project-compliance)
+* [Secure development practices](#secure-development-practices)
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|   |  |
+| -- | -- |
```
Comment (ragashreeshekar) [inline assessments/projects/rook/self-assessment.md:24]:
```suggestion
| -- | -- |
| Assessment Stage | Incomplete | 
```

**Review by JustinCappos — APPROVED** [review]
I approve.  Ready to merge.
